src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/InternationalPostalService.java

```diff
@@ -0,0 +1,216 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.ContactDetailsWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.IntervenerFourWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.IntervenerOneWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.IntervenerThreeWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.IntervenerTwoWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.intevener.IntervenerWrapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_RESIDE_OUTSIDE_UK;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERVENER1;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERVENER2;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERVENER3;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERVENER4;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_RESIDE_OUTSIDE_UK;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.IntervenerConstant.INTERVENER_FOUR;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.IntervenerConstant.INTERVENER_ONE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.IntervenerConstant.INTERVENER_THREE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.IntervenerConstant.INTERVENER_TWO;
+
+@Service
+@RequiredArgsConstructor
+public class InternationalPostalService {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * Validates the given FinremCaseData for any errors related to contact details.
+     *
+     * @param caseData The FinremCaseData to validate.
+     * @return A list of error messages. An empty list indicates no errors were found.
+     */
+    public List<String> validate(FinremCaseData caseData) {
+        ContactDetailsWrapper wrapper = caseData.getContactDetailsWrapper();
+        YesOrNo applicantResideOutsideUK = wrapper.getApplicantResideOutsideUK();
+        List<String> errors = new ArrayList<>();
+        if (ObjectUtils.isNotEmpty(applicantResideOutsideUK)
+            && applicantResideOutsideUK.equals(YesOrNo.YES)
+            && wrapper.getApplicantAddress() != null
+            && ObjectUtils.isEmpty(wrapper.getApplicantAddress().getCountry())) {
+            errors.add("If applicant resides outside of UK, Please provide the country of residence.");
```

Comment:
"Please" doesn't need a capital P (there are few other instances of this in the class)

Reply:
done